torchgeo/datasets/eurosat.py

```diff
@@ -318,3 +318,24 @@ class EuroSAT100(EuroSAT):
         'val': '3e3f8b3c344182b8d126c4cc88f3f215',
         'test': 'f908f151b950f270ad18e61153579794',
     }
+
+
+class SpatialSplitEuroSAT(EuroSAT):
```

Comment:
I wonder if EuroSATSpatial or EuroSATGeo would be a better name. It's up to you guys though, it's your dataset.

Another option would be to create a `GeoDataset` for `EuroSAT` that allows users to define their own geographic split. This would provide optimal flexibility, and allow you to combine with other imagery as well.

These are both completely optional. I don't know what the best approach is. It's probably fine to merge as is, just thinking outside the box to make sure we've covered every possibility.

Reply:
Great ideas! I will go with EuroSATSpatial then; it sounds a bit more descriptive, I guess.

Regarding refactoring as GeoDataset: the flexibility of defining custom geo splits sounds intriguing and could theoretically be applied to any of the NonGeoDatasets. However, I am not exactly sure if it's worth the extra code and maintenance. Overriding the splits sounds very convenient as I see it

Reply:
> and could theoretically be applied to any of the NonGeoDatasets

Only the ones that store files with geospatial metadata. Many use .jpg or .png and don't release this metadata.

If you do go with a GeoDataset, then EuroSATGeo might be a better name. That's what we did for LandCover.ai which has both Geo- and NonGeoDataset versions.

Reply:
Right, the ones with geotiff should be suitable only. Re GeoDataset: I would go with the current settings for now if that is fine. Would love to reformulate that way if I need higher flexibility (i.e., another set of partitions)